core/app/models/spree/promotion.rb

```diff
@@ -78,9 +97,21 @@ def activate(payload)
       return action_taken
     end
 
+    # Failsafe - backwards compatibility
+    def code
+      codes.first.try(:value)
+    end
+
+    # Failsafe - backwards compatibility
+    def code=(value)
```

Comment:
This is not backwards compatible, since it builds a new code every time.

You should also add a deprecation warning here.